### PR TITLE
Implement full signOut flow

### PR DESCRIPTION
**Pull request by thchia**
For #7

- Refactored Callback component to be generic and call whatever redirect function it was given.
- Added a SignoutCallback component, to be rendered at the appropriate signout url.
- Call signoutRedirect in the signOut method.

This will be released in the next **0.x** release, since the implementation of `signOut` has changed.

### CODE REVIEW

File: src/SignInCallback/index.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react'
+
+import Callback, { ICallbackActionProps } from '../Callback'
+
+const SignInCallback = (props: ICallbackActionProps) => {
+  return (
+    <Callback
+      onSuccess={props.onSuccess}
+      onError={props.onError}
+      redirectCallback={props.userManager.signinRedirectCallback}
```
Comment (thchia) [inline src/SignInCallback/index.tsx:10]: Need to `props.userManager.signinRedirectCallback.bind(props.userManager)`

File: src/SignOutCallback/index.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react'
+
+import Callback, { ICallbackActionProps } from '../Callback'
+
+const SignOutCallback = (props: ICallbackActionProps) => {
+  return (
+    <Callback
+      onSuccess={props.onSuccess}
+      onError={props.onError}
+      redirectCallback={props.userManager.signoutRedirectCallback}
```
Comment (thchia) [inline src/SignOutCallback/index.tsx:10]: See above.

**Comment by dankremniov** [comment]
@thchia, is there any reason why you closed this PR? Full sign out changes made to `src/makeAuth/index.tsx` makes total sense for me - exactly what I was looking for. I think to extend Authentificator with these changes locally. Just wondering if you have identified any blocking issue with that?

**Comment by thchia** [comment]
Hey, you can look through #7 for an explanation, but basically I determined that the functionality that I was trying to add was not compatible with all Identity Providers.

Since this is supposed to be IdP agnostic I closed the PR. However there are 2 ways for you to handle this:

- If your IdP supports this flow you can make any changes you want in your own project to support it. I don’t think there’s anything blocking it. 

- If you can implement a safe way to add this functionality to the library, feel free to make a PR :)

**Pull request closed without merging**